src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -2502,6 +2502,21 @@ bool QgsPostgresProvider::addFeatures( QgsFeatureList &flist, Flags flags )
         {
           v = paramValue( value.toString(), defaultValues[ i ] );
 
+          // the conversion functions expects the list as a string, so convert it
+          if ( value.type() == QVariant::StringList )
+          {
+            QStringList list_vals = value.toStringList();
+            // all strings need to be double quoted to allow special postgres
+            // array characters such as {, or whitespace in the string
+            // but we need to escape all double quotes
+            list_vals.replaceInStrings( "\"", "\\\"" );
```

Comment:
Shouldn't slashes be escaped too ? (before doing the double quote escaping)

Reply:
yes, correct, thank you